src/katgpucbf/xbgpu/bsend.py

```diff
@@ -172,10 +172,18 @@ def send(
 
             Also update its relevant counters and sensor values.
         """
-        if any(send_stream.tx_enabled):
+        n_enabled = sum(send_stream.tx_enabled)
+        rate = send_stream.bytes_per_second_per_beam * n_enabled
+        if n_enabled > 0:
             send_futures = []
             for frame in self._frames:
-                heaps_to_send = [heap for heap, enabled in zip(frame.heaps, send_stream.tx_enabled) if enabled]
+                # TODO: building this list every time may be too expensive.
+                # Consider caching it and invalidating when streams are enabled/disabled.
```

Comment:
I might have to request you create a ticket for this, please.

Reply:
Done.